mod/item.php

```diff
@@ -579,21 +483,10 @@ function item_post(App $a) {
 		$attaches = $match[1];
 		if (count($attaches)) {
 			foreach ($attaches as $attach) {
-				$r = q("SELECT * FROM `attach` WHERE `uid` = %d AND `id` = %d LIMIT 1",
-					intval($profile_uid),
-					intval($attach)
-				);
-				if (DBM::is_result($r)) {
-					$r = q("UPDATE `attach` SET `allow_cid` = '%s', `allow_gid` = '%s', `deny_cid` = '%s', `deny_gid` = '%s'
-						WHERE `uid` = %d AND `id` = %d",
-						dbesc($str_contact_allow),
-						dbesc($str_group_allow),
-						dbesc($str_contact_deny),
-						dbesc($str_group_deny),
-						intval($profile_uid),
-						intval($attach)
-					);
-				}
+				$fields = ['allow_cid' => $str_contact_allow, 'allow_gid' => $str_group_allow,
+						'deny_cid' => $str_contact_deny, 'deny_gid' => $str_group_deny];
+				$condition = ['id' => $attach];
```

Comment:
Why did you drop the `uid` condition?

Reply:
The primary key is already the "id" field. Querying an additional field doesn't make any sense.

Reply:
It does to make sure the attachment is owned by `$profile_uid`, can you ensure it is?